scripts/list_contracts.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/python3
+
+
+from itertools import groupby
+
+import click
+from ape.cli import ConnectedProviderCommand
+
+from deployment.constants import SUPPORTED_TACO_DOMAINS
+from deployment.registry import read_registry
+from deployment.utils import get_chain_name, registry_filepath_from_domain
+
+
+def format_chain_name(chain_name):
+    """
+    Format the chain name to capitalize each word and join with slashes.
+    """
+    return "/".join(word.capitalize() for word in chain_name.split())
+
+
+def get_registry_entries(selected_domain):
+    """
+    Parse the registry files for the given domain or all supported domains.
+    """
+    registry_entries = []
+
+    for domain in SUPPORTED_TACO_DOMAINS:
+        if selected_domain and selected_domain != domain:
+            continue
+        registry_filepath = registry_filepath_from_domain(domain=domain)
+        entries = read_registry(filepath=registry_filepath)
+        registry_entries.append((domain, entries))
+
+    return registry_entries
+
+
+def display_registry_entries(registry_entries):
+    """
+    Display registry entries grouped by chain ID.
+    """
+    for domain, entries in registry_entries:
+        grouped_entries = groupby(entries, key=lambda entry: entry.chain_id)
+        click.secho(f"\n{domain.capitalize()} Domain", fg="green")
+
+        for chain_id, chain_entries in grouped_entries:
+            chain_name = format_chain_name(get_chain_name(chain_id))
+            click.secho(f"{chain_name}", fg="yellow")
+
+            for index, entry in enumerate(chain_entries, start=1):
+                click.secho(f"    {index}. {entry.name} {entry.address}", fg="cyan")
```

Comment:
Wdyt about tabbing both of these:

```suggestion
            click.secho(f"    {chain_name}", fg="yellow")

            for index, entry in enumerate(chain_entries, start=1):
                click.secho(f"        {index}. {entry.name} {entry.address}", fg="cyan")
```

Leads to:
```
Lynx Domain
    Ethereum/Sepolia
        1. LynxStakingToken 0x347370278531Db455Aec3BFD0F30d57e41422353
        2. MockPolygonRoot 0xDD60a8E632c13fb777Ca76C7FE5670031202Edab
        3. TACoApplication 0x329bc9Df0e45f360583374726ccaFF003264a136
        4. TestnetThresholdStaking 0x18006f9A84C0bAD4CD96Aa69C7cE17aD760cDaD2
    Polygon/Amoy
        1. BetaProgramInitiator 0xf47dde316D994a050b8b4e5986e0790309979697
        2. Coordinator 0xE9e94499bB0f67b9DBD75506ec1735486DE57770
        3. GlobalAllowList 0xfDBA7100B015586270B62bA116920b78F4ff6930
        4. LynxRitualToken 0x064Be2a9740e565729BC0d47bC616c5bb8Cc87B9
        5. MockPolygonChild 0x4FD23FAB4A09F85872bf240ABBd484cb4F9a5F79
        6. OpenAccessAuthorizer 0x33270a0B88d0Ffb6B0b4FBA119ca6a7263DeF675
        7. SubscriptionManager 0x811389558a2C0B65ff56652d5E5bBF5DbC9A4358
        8. TACoChildApplication 0x42F30AEc1A36995eEFaf9536Eb62BD751F982D32

Tapir Domain
...
 ```

Reply:
Looks great! ✔️